presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergPlanOptimizer.java

```diff
@@ -125,6 +126,9 @@ public PlanNode visitFilter(FilterNode filter, RewriteContext<Void> context)
             }
 
             TableScanNode tableScan = (TableScanNode) filter.getSource();
+            if (((IcebergTableHandle) tableScan.getTable().getConnectorHandle()).getIcebergTableName().getTableType() != DATA) {
+                return visitPlan(filter, context);
+            }
```

Comment:
Should we reserve the table type of `EQUALITY_DELETES` and `DATA_WITHOUT_EQUALITY_DELETES`? Intuitively they are both partitioning aware, so might fit this optimization.

Reply:
Just to confirm, you mean that this rule should also run for the `EQUALITY_DELETES` and `DATA_WITHOUT_EQUALITY_DELETES`?

Reply:
Yes, but I'm not so sure, need to further confirm, I just bring it up for discussion. So, we can filter out them now and add them back when we confirm they are suitable, or we can just reserve them now and discard them when we find they are not suitable here. Which one do you prefer?

Reply:
I think we should confirm that they are suitable for this rule before allowing them. If we haven't confirmed yet, then let's exclude it until it is confirmed

Reply:
Sure, that's no problem.